src/overlays/actors/ovl_En_Niw/z_en_niw.c

```diff
@@ -21,61 +48,1192 @@ const ActorInit En_Niw_InitVars = {
+    EnNiw* this = THIS;
+    s64 pad;
+    Vec3f* posListEntry;
+    s32 i;
+
+    if (this->actor.params < 0) {
+        this->actor.params = 0;
+    }
+
+    if (this->actor.params == 0x000B) {
+        if (sSpawnedVer1 != 0) {
+            Actor_Kill(&this->actor);
+            osSyncPrintf("\x1b[33m☆☆☆☆☆ もういてる原 Ver.1 ☆☆☆☆☆ \n\x1b[m");
+            return;
+        }
+        sSpawnedVer1 = 1;
+        this->actor.room = -1;
+    }
+
+    if (this->actor.params == 0x000C) {
+        if (sSpawnedVer2 != 0) {
+            Actor_Kill(&this->actor);
+            osSyncPrintf("\x1b[33m☆☆☆☆☆ もういてる原 Ver.2 ☆☆☆☆☆ \n\x1b[m");
+            return;
+        }
+        sSpawnedVer2 = 1;
+        this->actor.room = -1;
+    }
+
+    Actor_ProcessInitChain(&this->actor, sInitChain);
+    this->actor.flags |= 1;
+    ActorShape_Init(&this->actor.shape, 0.0f, &ActorShadow_DrawFunc_Circle, 25.0f);
+    SkelAnime_InitSV(globalCtx, &this->skelAnime, &D_06002530, &D_060000E8, this->limbDrawTable,
+                     this->transitionDrawTable, 16);
+
+    if (globalCtx->sceneNum == SCENE_SPOT01) {
+        for (i = 0, posListEntry = sKakarikoPosList; i != 7; i++, posListEntry++) {
+            if (fabsf(this->actor.posRot.pos.x - posListEntry->x) < 40.0f &&
+                fabsf(this->actor.posRot.pos.z - posListEntry->z) < 40.0f) {
+                this->unk_2AA = i;
+                osSyncPrintf("\x1b[33m 通常鶏index %d\n\x1b[m", this->unk_2AA);
+                if (gSaveContext.infTable[25] & sKakarikoFlagList[i]) {
+                    this->actor.posRot.pos.x = 300.0f;
+                    this->actor.posRot.pos.y = 100.0f;
+                    this->actor.posRot.pos.z = 1530.0f;
+                    this->actor.params = 0x0000;
+                }
+            }
+        }
+    }
+
+    Math_Vec3f_Copy(&this->unk_2AC, &this->actor.posRot.pos);
+    Math_Vec3f_Copy(&this->unk_2B8, &this->actor.posRot.pos);
+    this->unk_304 = 10.0f;
+    Actor_SetScale(&this->actor, 0.01f);
+    this->unk_2A4 = (s16)Math_Rand_ZeroFloat(3.99f) + 5;
+
+    if (this->unk_2A4 < 0) {
+        this->unk_2A4 = 1;
+    }
+
+    switch (this->actor.params) {
+        case 2:
+            if (gSaveContext.nightFlag == 0) {
+                Actor_Kill(&this->actor);
+            }
+            break;
+        case 1:
+            if ((gSaveContext.eventChkInf[1] & 0x10) != 0) {
+                Actor_Kill(&this->actor);
+            }
+            break;
+        case 3:
+            if ((gSaveContext.eventChkInf[1] & 0x10) == 0) {
+                Actor_Kill(&this->actor);
+            }
+            break;
+        case 5:
+            if ((gSaveContext.eventChkInf[1] & 0x100) != 0) {
+                Actor_Kill(&this->actor);
+            }
+            break;
+        case 7:
+            if ((gSaveContext.eventChkInf[1] & 0x100) == 0) {
+                Actor_Kill(&this->actor);
+            }
+            break;
+        case 13:
```

Comment:
the switch is on `this->actor.params`, according to the style you went with elsewhere this `13` should be hexadecimal `0x000D` (and in other places in the file too)

not a fan of left padding with zeros, and I don't think using hex makes a lot of sense. The variable only seems to be different behaviors, without any "packed values" like `params & 0x3F == switch flag` in some actors

it's however you want but I would use decimal everywhere without 0-padding (at least, be consistent!)